python/triton/compiler/compiler.py

```diff
@@ -218,6 +218,7 @@ def compile(src, target=None, options=None):
     if target is None:
         target = driver.active.get_current_target()
     backend = make_backend(target)
+    ir_source = not isinstance(src, IRSource)
```

Comment:
I don't follow the logic here. `ir_source` is true when `src` *is not* instance of `IRSource`?

Reply:
yeah I messed up something, let me try again

Reply:
ok, bad autofill... can you check again?

Reply:
LGTM!